common/types/mapper/thrift/history.go

```diff
@@ -2514,7 +2514,7 @@ func FromRecordActivityTaskStartedResponse(t *types.RecordActivityTaskStartedRes
 	return &history.RecordActivityTaskStartedResponse{
 		ScheduledEvent:                  FromHistoryEvent(t.ScheduledEvent),
 		StartedTimestamp:                t.StartedTimestamp,
-		Attempt:                         t.Attempt,
+		Attempt:                         &t.Attempt,
```

Comment:
same "somewhat concerned about stored data" note, but a 0 here makes sense too even if it changes from not-stored to stored.  so probably not a problem either way.